corehq/apps/users/middleware.py

```diff
@@ -50,7 +50,12 @@ def process_view(self, request, view_func, view_args, view_kwargs):
         if request.user.is_anonymous and 'domain' in view_kwargs:
             if ANONYMOUS_WEB_APPS_USAGE.enabled(view_kwargs['domain']):
                 request.couch_user = CouchUser.get_anonymous_mobile_worker(request.domain)
-        if determine_authtype_from_header(request) in (BASIC, DIGEST,) and 'domain' in view_kwargs:
+        if (
+            'domain' in view_kwargs and (
+                determine_authtype_from_header(request) in (BASIC, DIGEST) or
+                # Support API requests that pass the username as a GET parameter
+                'username' in request.GET)
+        ):
```

Comment:
@kaapstorm it looks like `determine_authtype_from_header` can return `API_KEY` as an option, which would be preferable, so there's only one way to determine what auth to use.